src/views/LeadAdd/LeadDetail/index.js

```diff
@@ -424,12 +424,13 @@ class LeadDetail extends React.PureComponent {
         const {
             requests: { fileUrlGetRequest },
             lead,
+            onLeadAttachmentChange,
         } = this.props;
 
         const values = leadFaramValuesSelector(lead);
         const { attachment } = values;
 
-        if (isDefined(attachment)) {
+        if (isDefined(onLeadAttachmentChange) && isDefined(attachment)) {
```

Comment:
This is not a proper way to handle this behavior.
This doesn't not answer why we are doing this.
We need a prop to enable/disable this behavior (eg: shouldFetchAttachment)
We should also check this prop on fileUrlGetRequest success